### PR TITLE
users: Fix handling of failed date parsing

### PR DESCRIPTION
**Pull request by chriswiggins**
Hi,

On systems without coreutils date (i.e busy box-based systems), the catch on the date parsing was causing an undefined entry to come back. This resolves that by returning last login as null

### CODE REVIEW

File: pkg/users/users.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ function AccountsPage() {
 
     useInit(() => {
         const debouncedGetLogins = debounce(100, () => {
-            getLogins().then(setDetails);
+            getLogins().then((logins) => setDetails(logins.filter((login) => !!login)));
```
Comment (martinpitt) [inline pkg/users/users.js:74]: This is a bit too late for my taste -- empty logins should rather be filtered out inside of getLogins() in the output parsing of the respective program.

Do you have an example `lastlog` or `w` or `passwd -S --all` output which causes that? I'd like to put that into our integration test to actually cover and verify this update.

File: pkg/users/users.js [hunk 2]
```diff
@@ -206,7 +206,7 @@ async function getLogins() {
 
         const loggedIn = currentLogins.includes(name);
 
-        const date_fields = splitLine.slice(-5);
+        const date_fields = splitLine.slice(4);
```
Comment (martinpitt) [inline pkg/users/users.js:209]: How does the `lastlog` output look like for a broken case? I'm very hesistant to introduce an assumption about the number of initial columns, as they are variable. Check the difference between a local and an SSH login:
```
mpitt            pts/6    ::1                                       Wed Apr 10 07:38:25 +0200 2024
martin           tty1                                               Sat Apr  6 19:20:11 +0200 2024
```
That's the reason why I picked the last 5 filelds to capture the date.

lastlog is really hard to machine read, it's a terrible output format. I wish it had a `--json` mode or something similar.. :cry: It's also not reasonable to read /var/log/lastlog with the cockpit/JS API. An alternative may be to spawn a python helper which reads the file and spits out JSON information, but as the [file format](https://manpagez.com/man/5/lastlog/) has platform/architecture specific record sizes this also seems hard.

So in the end we are stuck with calling `lastlog` :cry: 

Please send the offending lastlog output, we can hopefully refine the heuristics to make it work.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! What a ratsnest..

File: pkg/users/users.js (hunk 2 above)
Comment (chriswiggins) [reply pkg/users/users.js:209]: OK! So I may have caused some confusion with this - my apologies 😞 

I was testing these two changes on two systems, one ubuntu and one yocto. Fixing the undefined issue hid the formatting issue in Yocto. Our Yocto build uses busybox `date`, which doesn't support the formatting options. We are going to switch that out for coreutils date so we can fix this. I will revert this change.

File: pkg/users/users.js (hunk 1 above)
Comment (chriswiggins) [reply pkg/users/users.js]: I've fixed this on the catch line at 212 - not returning anything from that was causing the undefined to show up. This at least fixes the issues on systems with non-coreutils date

**Comment by chriswiggins** [comment]
Apologies for the run-around. It was just causing issues on our Yocto system without coreutils date, which is why the undefined was showing up, not the positioning of the splitting.

I will update the title and description accordingly 👍

**Comment by martinpitt** [comment]
Ooh, that's a good fix, thanks! I'd like to factorize this a bit though, do you mind if I just force-push to your branch, squash the commits, fix the changelog and let you do a final review?

File: pkg/users/users.js [hunk 3]
```diff
@@ -210,7 +210,12 @@ async function getLogins() {
         // this is impossible to parse with Date() (e.g. Firefox does not work with all time zones), so call `date` to parse it
         return cockpit.spawn(["date", "+%s", "-d", date_fields.join(' ')], { environ: ["LC_ALL=C"], err: "out" })
                 .then(out => ({ name, loggedIn, lastLogin: parseInt(out) * 1000, isLocked }))
```
Comment (martinpitt) [inline pkg/users/users.js:212]: If you want to do that yourself: Just record the date from the output here, have the `catch()` branch set it to `null`, and return a single result object.

Comment (chriswiggins) [reply pkg/users/users.js:212]: Happy for you to do it! I'm not quite sure I'm understanding what you mean, so if you push I'll check

Comment (martinpitt) [reply pkg/users/users.js]: Done -- can you please have a look? (Tests are still running)

Comment (chriswiggins) [reply pkg/users/users.js]: LGTM! 👍

**Review by martinpitt — APPROVED** [review]
Great, thanks for the cross-check! If that goes green, I'm happy. Thanks for spotting and fixing!

**Pull request merged**